packages/playback-core/src/index.ts

```diff
@@ -554,6 +566,20 @@ export const loadMedia = (
       },
       { once: true }
     );
+    addEventListenerWithTeardown(mediaEl, 'pause', () => {
+      if (!getEnded(mediaEl)) return;
+      if (mediaEl.ended) return;
+      // This means we've "pseudo-ended". Dispatch an event to notify the outside world.
+      mediaEl.dispatchEvent(new Event('ended'));
+    });
+
+    addEventListenerWithTeardown(mediaEl, 'play', () => {
+      if (mediaEl.ended) return;
+      if (!isApproximatelyGTE(mediaEl.currentTime, mediaEl.duration)) return;
+      // If we were "pseudo-ended" before playback was attempted, seek back to the
+      // beginning to "replay", like "real" ended behavior.
+      mediaEl.currentTime = mediaEl.seekable.start(0);
+    });
```

Comment:
this could also be easily added to the React versions but yeah I would be fully on board to make those wrappers around mux-video and mux-audio